test/unit/transitions/multi_form_alerts.js

```diff
@@ -0,0 +1,384 @@
+    test.ok(err);
+    test.done();
+  });
+};
+
+exports['validates config : isReportCounted'] = test => {
+  testValidate(test, _.omit(alert, 'isReportCounted'));
+};
+
+exports['validates config : numReportsThreshold'] = test => {
+  testValidate(test, _.omit(alert, 'numReportsThreshold'));
+};
+
+exports['validates config : message'] = test => {
+  testValidate(test, _.omit(alert, 'message'));
+};
+
+exports['validates config : recipients'] = test => {
+  testValidate(test, _.omit(alert, 'recipients'));
+};
+
+exports['validates config : timeWindowInDays'] = test => {
+  testValidate(test, _.omit(alert, 'timeWindowInDays'));
+};
+
+exports['fetches reports within time window'] = test => {
+  sinon.stub(config, 'get').returns([alert]);
+
+  transition.onMatch({ doc: doc }, undefined, undefined, () => {
+    test.equals(utils.getReportsWithinTimeWindow.callCount, 1);
+    test.equals(utils.getReportsWithinTimeWindow.args[0][0], 12344);
+    test.equals(utils.getReportsWithinTimeWindow.args[0][1], alert.timeWindowInDays);
+    test.done();
+  });
+};
+
+exports['filters reports by form if forms is present in config'] = test => {
+  sinon.stub(config, 'get').returns([_.extend({forms: ['A']} , alert)]);
+
+  transition.onMatch({ doc: doc }, undefined, undefined, () => {
+    const countedReports = vm.runInNewContext.getCalls().map((call) => call.args[1].report);
+    test.equals(countedReports.length, 2);
+    test.equals(countedReports[0]._id, doc._id);
+    test.equals(countedReports[1]._id, reports[0]._id);
+    test.done();
+  });
+};
+
+exports['evals isReportCounted'] = test => {
+  sinon.stub(config, 'get').returns([alert]);
+
+  transition.onMatch({ doc: doc }, undefined, undefined, () => {
+    const script = vm.runInNewContext.args[0][0];
+    test.ok(script.includes(alert.isReportCounted));
+    test.done();
+  });
+};
+
+exports['if not enough reports pass the isReportCounted func, does nothing'] = test => {
+  sinon.stub(config, 'get').returns([alert]);
+
+  // make all reports fail
+  vm.runInNewContext.restore();
+  sinon.stub(vm, 'runInNewContext').returns(false);
+
+  transition.onMatch({ doc: doc }, undefined, undefined, () => {
+    test.equals(vm.runInNewContext.getCalls().length, alert.numReportsThreshold); // 3
+    test.equals(messages.addError.getCalls().length, 0);
+    test.equals(messages.addMessage.getCalls().length, 0);
+    test.done();
+  });
+};
+
+exports['if no reports in time window, does nothing'] = test => {
+  sinon.stub(config, 'get').returns([alert]);
+
+  utils.getReportsWithinTimeWindow.restore();
+  sinon.stub(utils, 'getReportsWithinTimeWindow').returns(Promise.resolve([]));
+
+  transition.onMatch({ doc: doc }, undefined, undefined, (err, docNeedsSaving) => {
+    test.equals(messages.addError.getCalls().length, 0);
+    test.equals(messages.addMessage.getCalls().length, 0);
+    test.ok(!docNeedsSaving);
+    test.done();
+  });
+};
+
+const assertMessage = (test, call, recipient, message) => {
```

Comment:
Modify this to take `messageArgs` not `call` so it doesn't have to know that they came from sinon.